File: client/src/app/modules/miscellaneous/services/provider.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Provider } from '../../miscellaneous/model/Provider';

@Injectable({
  providedIn: 'root'
})
export class ProviderService extends AppRestService<Provider> {

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getProviders() {
    return this.getList(new HttpParams(), "providers");
  }

  getProviderByValue(value: string) {
    return this.getList(new HttpParams().set("value", value), "providers/search");
  }

  addOrUpdateProvider(provider: Provider) {
    return this.addOrUpdate(new HttpParams(), "provider", provider, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/ProviderService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.libs.exception.OsirisException;
import com.jss.osiris.modules.osiris.miscellaneous.model.Provider;

public interface ProviderService {
    public List<Provider> getProviders();

    public Provider getProvider(Integer id);

    public Provider addOrUpdateProvider(Provider provider) throws OsirisException;

    public List<Provider> getProviderByValue(String upperCase);
}
